kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/database/Db.kt

```diff
@@ -0,0 +1,66 @@
+package xyz.block.ftl.database
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.exc.MismatchedInputException
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import xyz.block.ftl.logging.Logging
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.sql.Connection
+import java.sql.DriverManager
+
+private const val DEFAULT_DB_YAML = "db.yaml"
+
+data class DatabaseConfig(
+  val databases: Map<String, Database>,
+)
+
+data class Database(
+  val jdbcUrl: String,
+  val username: String,
+  val password: String,
+)
+
+/**
+ * `Db` is a simple wrapper around the JDBC driver manager that provides a connection to the database specified
+ * by the DB YAML file (`db.yaml` unless otherwise specified). The YAML must be provided in the `src/main/resources`
```

Comment:
Switch this to use an environment variable like `FTL_POSTGRES_DSN_EXAMPLE1=postgresql://postgres-10.4.cluster:5432/default?user=postgres&password=postgres` for this rather than a config file.

It makes local development much simpler, and additionally it will have to be dynamically constructed in production, so putting it into `src/main/resources` will be difficult vs. just using an envar. Also this needs to be cross-language compatible, so it can't include a prefix like `jdbc`, just the raw DSN.